src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -1713,6 +1713,59 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                                             INS_SCALABLE_OPTS_UNPREDICATED);
                 break;
 
+            case NI_Sve_SaturatingDecrementBy16BitElementCount:
+            case NI_Sve_SaturatingDecrementBy32BitElementCount:
+            case NI_Sve_SaturatingDecrementBy64BitElementCount:
+            case NI_Sve_SaturatingDecrementBy8BitElementCount:
+            case NI_Sve_SaturatingIncrementBy16BitElementCount:
+            case NI_Sve_SaturatingIncrementBy32BitElementCount:
+            case NI_Sve_SaturatingIncrementBy64BitElementCount:
+            case NI_Sve_SaturatingIncrementBy8BitElementCount:
+            {
+                assert(isRMW);
+                if (targetReg != op1Reg)
+                {
+                    GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op1Reg, /* canSkip */ true);
+                }
+
+                // Cannot use create a lookup table for 2 immediates. Therefore the immediates must be constants,
+                // and can be used directly (instead of via HWIntrinsicImmOpHelper).
+
+                assert(hasImmediateOperand);
+                int           scale   = (int)intrin.op2->AsIntCon()->gtIconVal;
+                insSvePattern pattern = (insSvePattern)intrin.op3->AsIntCon()->gtIconVal;
+
+                // Use scalable vector sizes.
+                GetEmitter()->emitIns_R_PATTERN_I(ins, emitSize, targetReg, pattern, scale, opt);
+                break;
+            }
+
+            case NI_Sve_SaturatingDecrementBy16BitElementCountScalar:
+            case NI_Sve_SaturatingDecrementBy32BitElementCountScalar:
+            case NI_Sve_SaturatingDecrementBy64BitElementCountScalar:
+            case NI_Sve_SaturatingIncrementBy16BitElementCountScalar:
+            case NI_Sve_SaturatingIncrementBy32BitElementCountScalar:
+            case NI_Sve_SaturatingIncrementBy64BitElementCountScalar:
+            {
+                assert(isRMW);
+                if (targetReg != op1Reg)
+                {
+                    GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op1Reg, /* canSkip */ true);
+                }
+
+                // Cannot use create a lookup table for 2 immediates. Therefore the immediates must be constants,
+                // and can be used directly (instead of via HWIntrinsicImmOpHelper).
+
+                assert(hasImmediateOperand);
+                int           scale   = (int)intrin.op2->AsIntCon()->gtIconVal;
+                insSvePattern pattern = (insSvePattern)intrin.op3->AsIntCon()->gtIconVal;
+
+                // Use scalar sizes.
+                emitSize = emitActualTypeSize(node->gtType);
```

Comment:
The only difference between these set of cases and the vector variant is this line. Can we combine the 2 sets by:

```c++
case *scalar:
   emitSize = emitActualTypeSize(node->gtType);
   FALLTHROUGH;
case *vector:
   // logic
```

Reply:
There is also the use of `opt` argument which needs to either be `INS_OPTS_NONE` or not passed for the scalar variant.

Reply:
... but that can be handled easily. Done.